plugins/source/mysql/client/types.go

```diff
@@ -1,64 +1,143 @@
 package client
 
 import (
-	"github.com/cloudquery/plugin-sdk/v2/schema"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/apache/arrow/go/v13/arrow"
+	"github.com/cloudquery/plugin-sdk/v3/types"
 )
 
-func SQLType(t schema.ValueType) string {
-	switch t {
-	case schema.TypeBool:
+const defaultPrecision = 10
+const defaultScale = 0
+
+func getValue(str string, defaultValue int32) int32 {
+	val, err := strconv.ParseInt(str, 10, 32)
+	if err != nil {
+		return defaultValue
+	}
+	return int32(val)
+}
+
+func getPrecisionAndScale(dataType string) (precision, scale int32) {
+	str := strings.TrimPrefix(dataType, "decimal")
+	str = strings.TrimPrefix(str, "numeric")
+	if str == "" {
+		return defaultPrecision, defaultScale
+	}
+	str = strings.TrimPrefix(str, "(")
+	str = strings.TrimSuffix(str, ")")
+	parts := strings.Split(str, ",")
+
+	switch len(parts) {
+	case 1:
+		precision = getValue(parts[0], defaultPrecision)
+		scale = defaultScale
+	case 2:
+		precision = getValue(parts[0], defaultPrecision)
+		scale = getValue(parts[1], defaultScale)
+	default:
+		precision = defaultPrecision
+		scale = defaultScale
+	}
+	return precision, scale
+}
+
+func SQLType(t arrow.DataType) string {
+	switch v := t.(type) {
+	case *arrow.BooleanType:
 		return "bool"
-	case schema.TypeInt:
```

Comment:
To avoid losing data for unsigned bigint I added support to all various int types + unsigned ones.
This is a breaking change